pkg/v7/configmap/configmap.go

```diff
@@ -42,12 +46,29 @@ func New(config Config) (*Service, error) {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Tenant must not be empty", config)
 	}
 
+	// Azure manages Calico CIDR blocks differently, and the installations
+	// settings can be empty.
+	if config.Provider != label.ProviderAzure {
```

Comment:
So why is the provider still relevant in the common packages? That does not make sense to me. We should actually try to change that.

Reply:
Agree. I'm working on the chartconfig resource and it will pass a ChartSpec to the chartconfig service. In the per provider resources any provider specific ChartSpecs are injected. So pkg only contains shared code.

I hope we can use the same approach for configmaps. I'll ping you both in the PR when its ready and we can discuss there.

Reply:
In this case some of the params required by coredns chart are empty in Azure, because Calico is not managed the same way as in AWS and KVM, see https://gigantic.slack.com/archives/C2MS2VB37/p1537337880000100 for details.